packages/blueprints-integration/src/triggers.ts

```diff
@@ -0,0 +1,282 @@
+	undo?: boolean
+}
+
+export interface IMoveNextAction extends ITriggeredActionBase {
+	action: PlayoutActions.moveNext
+	filterChain: (IRundownPlaylistFilterLink | IGUIContextFilterLink)[]
+
+	/**
+	 * "Vertical delta" - the next Part will be moved to the beginning of the specified segment
+	 *
+	 * @type {number}
+	 * @memberof IMoveNextAction
+	 */
+	segments: number
+	/**
+	 * "Horizontal delta" - the next Part will be moved by this amount within the selected segment
+	 *
+	 * @type {number}
+	 * @memberof IMoveNextAction
+	 */
+	parts: number
+}
+
+export interface ICreateSnapshotForDebugAction extends ITriggeredActionBase {
+	action: PlayoutActions.createSnapshotForDebug
+	filterChain: (IRundownPlaylistFilterLink | IGUIContextFilterLink)[]
+}
+
+export interface IDisableNextPieceAction extends ITriggeredActionBase {
+	action: PlayoutActions.disableNextPiece
+	filterChain: (IRundownPlaylistFilterLink | IGUIContextFilterLink)[]
+	undo?: boolean
+}
+
+export interface IRundownPlaylistReloadDataAction extends ITriggeredActionBase {
+	action: PlayoutActions.reloadRundownPlaylistData
+	filterChain: (IRundownPlaylistFilterLink | IGUIContextFilterLink)[]
+}
+
+export interface IRundownPlaylistResetAction extends ITriggeredActionBase {
+	action: PlayoutActions.resetRundownPlaylist
+	filterChain: (IRundownPlaylistFilterLink | IGUIContextFilterLink)[]
+}
+
+export interface IRundownPlaylistResyncAction extends ITriggeredActionBase {
+	action: PlayoutActions.resyncRundownPlaylist
+	filterChain: (IRundownPlaylistFilterLink | IGUIContextFilterLink)[]
+	force?: boolean
+}
+
+export interface IShelfAction extends ITriggeredActionBase {
+	action: ClientActions.shelf
+	state: true | false | 'toggle'
+	filterChain: IGUIContextFilterLink[]
+}
+
+export interface IGoToOnAirLineAction extends ITriggeredActionBase {
+	action: ClientActions.goToOnAirLine
+	filterChain: IGUIContextFilterLink[]
+}
+
+export interface IRewindSegmentsAction extends ITriggeredActionBase {
+	action: ClientActions.rewindSegments
+	filterChain: IGUIContextFilterLink[]
+}
+
+export interface IShowEntireCurrentSegmentAction extends ITriggeredActionBase {
+	action: ClientActions.showEntireCurrentSegment
+	filterChain: IGUIContextFilterLink[]
+	on: boolean
+}
+
+export type SomeAction =
+	| IAdlibPlayoutAction
+	| IRundownPlaylistActivateAction
+	| IRundownPlaylistDeactivateAction
+	| ITakeAction
+	| IHoldAction
+	| IMoveNextAction
+	| ICreateSnapshotForDebugAction
+	| IDisableNextPieceAction
+	| IRundownPlaylistReloadDataAction
+	| IRundownPlaylistResetAction
+	| IRundownPlaylistResyncAction
+	| IShelfAction
+	| IGoToOnAirLineAction
+	| IRewindSegmentsAction
+	| IShowEntireCurrentSegmentAction
+
+export interface IBlueprintTriggeredActions {
+	_id: string
+	/** Rank number for visually ordering the hotkeys */
+	_rank: number
+	/** Optional label to specify what this triggered action is supposed to do, a comment basically */
+	name?: ITranslatableMessage | string
```

Comment:
I don't think this should be translatable, assuming these are only shown to the person configuring the triggers. It will add complexity to the ui and import/export, for very little benefit.
I am curious to know your thoughts

Reply:
No, in some cases (like system-wide/core action triggers), this name will be displayed to the User in the Help panel in the Shelf. So this is the name that will be displayed for triggers like "Take", etc. Most of these will be provided by Core migrations, but the Blueprints can override that.

Reply:
Hmm ok
Will that work with the translations namespacing?

So it sounds like it is needed to be able to translate some triggers, but that is going to require all triggers to be defined in the blueprints and will make them much harder to create in the ui when only wanted in a single installation
I guess that for now we should leave it as being translatable, and we can figure out a solution when/if we start havign a problem

Reply:
I think theoretically it should, or at least, is in no way in a worse position to work than any other blueprint-provided translations for any other objects. Core-side, ITranslatableMessage has the namespace field which specifies the proper namespace (blueprint) to be used for it. Should still work, and if it doesn't, then it's just a bug in the processing done by Core when Blueprints are setting those values.

I assume that if they are only wanted in a single installation, then the solution is to just manually enter the label in a single/"universal" language. I think the need to be able to handle multiple languages for manually created Triggers is a pretty niche one.

Reply:
> I think theoretically it should, or at least, is in no way in a worse position to work than any other blueprint-provided translations for any other objects

I was thinking that if core creates a action trigger, then a blueprint migration changes the text of it, then can we figure out what namespace to translate it under? It feels like we should, its an edge case to be careful of.

> I assume that if they are only wanted in a single installation, then the solution is to just manually enter the label in a single/"universal" language.

Yeah that will probably be good enough. And if not then there is a clear solution of defining it via a blueprint migration.